modules/pubxaiAnalyticsAdapter.js

```diff
@@ -132,6 +103,36 @@ export function getDeviceType() {
   return 'desktop';
 }
 
+export function getBrowser() {
+  var isChrome =
+    /Chrome/.test(navigator.userAgent) && /Google Inc/.test(navigator.vendor);
+  var isCriOS = navigator.userAgent.match('CriOS');
+  var isSafari =
+    /Safari/.test(navigator.userAgent) &&
+    /Apple Computer/.test(navigator.vendor);
+  var isFirefox = /Firefox/.test(navigator.userAgent);
+  var isIE =
+    /Trident/.test(navigator.userAgent) || /MSIE/.test(navigator.userAgent);
+  var isEdge = /Edge/.test(navigator.userAgent);
+  if (isIE) return 'Internet Explorer';
+  if (isEdge) return 'Microsoft Edge';
+  if (isCriOS) return 'Chrome';
+  if (isSafari) return 'Safari';
+  if (isFirefox) return 'Firefox';
+  if (isChrome) return 'Chrome';
+  return 'Others';
```

Comment:
Instead of assigning variables for every single combo, I think it would be cleaner to just do a chain of if / else ifs (and maybe sort them in a order of % of users who use which browser even to make it even more optimized!

```js
if (/Chrome/.test(navigator.userAgent) && /Google Inc/.test(navigator.vendor)) return 'Chrome';
else if (navigator.userAgent.match('CriOS')) return 'Chrome';
else if (/Firefox/.test(navigator.userAgent)) return 'Firefox';
else if (/Safari/.test(navigator.userAgent) && /Apple Computer/.test(navigator.vendor)) return 'Safari';
else if (/Edge/.test(navigator.userAgent)) return 'Microsoft Edge';
else if (/Trident/.test(navigator.userAgent) || /MSIE/.test(navigator.userAgent)) return 'Internet Explorer';
else return 'Others';
```

Maybe something like that?

My best guess is Chrome > FF > Safari > Edge > IE

But feel free to order them however you would like.

Reply:
This function was taken directly from another adapter and I kept it as it is. Thanks for your inputs, I have updated the code as you suggested.

Reply:
This seems wrong, order matters,  you have to test for edge before safari for example and your edge test is wrong

https://docs.microsoft.com/en-us/microsoft-edge/web-platform/user-agent-string

Reply:
@patmmccann I have updated the order so that edge test comes before safari and updated edge token to `Edg` instead of `Edge`. Thanks !

Reply:
@patmmccann 

Sorry, was not aware that order mattered. Which is why I ordered by what I thought was most used browsers!

Makes sense if both have similar UA's